core/node/rules/can_create_stream.go

```diff
@@ -242,6 +242,7 @@ func (ru *csParams) canCreateStream() ruleBuilderCS {
 			).
 			requireMembership(
 				inception.ChannelId,
+				inception.SpaceId,
```

Comment:
if spaceId is nil (which is the case when posting images to dm's) the check for RequiredMemberships is going to fail. Looks like we should add a test for uploading media to a dm.

Reply:
this is handled in the change in `rule_builder.go` in this PR. Take a look and lmk if the change needs more tweaks.
There are two existing media tests that verify creating the media for dm and gdm with nil spaceId. In media.test.ts:
- userCanUploadMediaToDmIfMember
- userCanUploadMediaToGdmIfMember

Reply:
Nice.